ndcube/tests/helpers.py

```diff
@@ -3,22 +3,74 @@
 Helpers for testing ndcube.
 """
 import unittest
+from pathlib import Path
+from functools import wraps
 
+import astropy
+import matplotlib as mpl
+import matplotlib.pyplot as plt
 import numpy as np
+import pytest
+import sunpy
 from astropy.wcs.wcsapi.fitswcs import SlicedFITSWCS, SlicedLowLevelWCS
 from astropy.wcs.wcsapi.low_level_api import BaseLowLevelWCS
 from astropy.wcs.wcsapi.sliced_low_level_wcs import sanitize_slices
 from numpy.testing import assert_equal
 
 from ndcube import NDCube, NDCubeSequence, utils
 
-__all__ = ['assert_extra_coords_equal',
+__all__ = ['figure_test',
+           'get_hash_library_name',
+           'assert_extra_coords_equal',
            'assert_metas_equal',
            'assert_cubes_equal',
            'assert_cubesequences_equal',
            'assert_wcs_are_equal']
 
 
+def get_hash_library_name():
+    """
+    Generate the hash library name for this env.
+    """
+    ft2_version = f"{mpl.ft2font.__freetype_version__.replace('.', '')}"
+    mpl_version = "dev" if "+" in mpl.__version__ else mpl.__version__.replace('.', '')
+    sunpy_version = "dev" if "dev" in sunpy.__version__ else sunpy.__version__.replace('.', '')
+    astropy_version = "dev" if "dev" in astropy.__version__ else astropy.__version__.replace('.', '')
+    return f"figure_hashes_mpl_{mpl_version}_ft_{ft2_version}_astropy_{astropy_version}_sunpy_{sunpy_version}.json"
+
+
+def figure_test(test_function):
+    """
+    A decorator for a test that verifies the hash of the current figure or the
+    returned figure, with the name of the test function as the hash identifier
+    in the library. A PNG is also created in the 'result_image' directory,
```

Comment:
Is the result_image a tmp dir or is it actually in the repo?

Reply:
This docstring is well out of date, will nuke.